app/controllers/application_controller/tree_support.rb

```diff
@@ -22,11 +22,15 @@ def squash_toggle
   end
 
   def find_record
-    if %w(container_image host).include? controller_name
-      identify_record(params[:id], controller_name.classify)
-    else
-      identify_record(params[:id], VmOrTemplate)
-    end
+    # TODO: This logic should probably be reversed - fixed list for VmOrTemplate.
+    # (Better yet, override the method only in VmOrTemplate related controllers.)
```

Comment:
The reason I'm not doing this is I'm not sure what's the exact list that should result in `VmOrTemplate`.
The original `controller_name == 'host' ? Host : VmOrTemplate` code dates back to the "Initial Commit", 
and `find_record` seems widely used.